sdk/ml/azure-ai-ml/azure/ai/ml/entities/_credentials.py

```diff
@@ -342,6 +343,29 @@ def _from_rest_object(cls, obj: RestJobIdentityConfiguration) -> "Identity":
             error_category=ErrorCategory.SYSTEM_ERROR,
         )
 
+    @classmethod
+    def _load(
+        cls,
+        data: Dict = None,
+    ) -> Union["ManagedIdentityConfiguration", "UserIdentityConfiguration", "AmlTokenConfiguration"]:
+        type_str = data.get(CommonYamlFields.TYPE)
+        if type_str == camel_to_snake(ConnectionAuthType.MANAGED_IDENTITY):
```

Comment:
why not do this in reverse, so less function calls

Reply:
Thanks! updated.